KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs

```diff
@@ -173,7 +183,91 @@ internal object GetContentItemModel(Type t, JToken item, JToken modularContent,
                 }
             }
 
+            foreach (var property in richTextPropertiesToBeProcessed)
+            {
+                var value = property.GetValue(instance);
+                var propValue = ((JObject)item["elements"]).Properties()
+                    ?.FirstOrDefault(p => PropertyMapper.IsMatch(property, p.Name, system?.Type))
+                    ?.FirstOrDefault()["value"];
+
+                var modularContentInRichText =
+                                ((JObject)propValue?.Parent?.Parent)?.Property("modular_content")?.Value;
+
+                var currentlyProcessedString = new RichTextContentItem()
+                {
+                    ContentItemCodeName = system.Codename,
+                    RichTextElementCodeName = property.Name
+                };
+                if (currentlyResolvedRichStrings.Contains(currentlyProcessedString))
+                {
+                    value = RemoveContentItemsFromRichText((string) value);     //In case we've stumbled upon an element which is already being processed, we need to use it as 
+                                                                                //is (therefore removing content items) to prevent circular dependency
+                }
+                else
+                {
+                    currentlyResolvedRichStrings.Add(currentlyProcessedString);
+                    value = ProcessContentItemsInRichText(modularContent, processedItems, (string) value,
+                        modularContentInRichText, currentlyResolvedRichStrings);
+                    currentlyResolvedRichStrings.Remove(currentlyProcessedString);
+                }
+                if (value != null)
+                {
+                    property.SetValue(instance, value);
+                }
+
+            }
+
             return instance;
         }
+
+        private object ProcessContentItemsInRichText(JToken modularContent, Dictionary<string, object> processedItems, string value, JToken modularContentInRichText, HashSet<RichTextContentItem> currentlyResolvedRichStrings)
```

Comment:
Technically even then they could write a resolver that would produce the same object tag if they needed to, and possibly also add extra attribute to that with serialized JSON of the item ;-)